src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvestingClientServiceBean.java

```diff
@@ -167,28 +167,20 @@ public void deleteClient(Long clientId) {
     
     @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
     public void setHarvestSuccess(Long hcId, Date currentTime, int harvestedCount, int failedCount, int deletedCount) {
-        HarvestingClient harvestingClient = em.find(HarvestingClient.class, hcId);
-        if (harvestingClient == null) {
-            return;
-        }
-        em.refresh(harvestingClient);
-        
-        ClientHarvestRun currentRun = harvestingClient.getLastRun();
-        
-        if (currentRun != null && currentRun.isInProgress()) {
-            // TODO: what if there's no current run in progress? should we just
-            // give up quietly, or should we make a noise of some kind? -- L.A. 4.4      
-            
-            currentRun.setSuccess();
-            currentRun.setFinishTime(currentTime);
-            currentRun.setHarvestedDatasetCount(new Long(harvestedCount));
-            currentRun.setFailedDatasetCount(new Long(failedCount));
-            currentRun.setDeletedDatasetCount(new Long(deletedCount));
-        }
+        recordHarvestJobStatus(hcId, currentTime, harvestedCount, failedCount, deletedCount, ClientHarvestRun.RunResultType.INTERRUPTED);
```

Comment:
Not sure I understand - is this a typo? the setHarvestSuccess method is using the RunResultType.INTERRUPTED?

Reply:
That was of course a typo, thank you. 
(I'm working on automated tests right now that would catch this)